controllers/api/v1/services.controller.js

```diff
@@ -1,80 +1,43 @@
 'use strict';
 
-const serviceModel = require('../../../models/v1/Service.model.js'),
-  routeModel = require('../../../models/v1/Route.model.js');
+const _ = require('lodash');
 
 module.exports = ServicesController;
 
 function ServicesController(options) {
   options = options || {};
   const logger = options.logger;
-  const ServiceModel = new serviceModel(options);
-  const RouteModel = new routeModel(options);
+  const ServiceModel = options.models.ServiceModel;
+  const RouteModel = options.models.RouteModel;
 
   this.getServices = function(req, res, next) {
     var services = [];
     var included = [];
+    const query = {
+      '$or': [
+        {public: true},
+        {clientId: res._service.clientId}
+      ]
+    };
+    const options = {
+      sort: {name: 1},
+      populate: {path: 'routes', model: RouteModel.io}
+    };
     ServiceModel
       .io
-      .find({
-        $or: [
-          {public: true},
-          {clientId: res._service.clientId}
-        ]
-      })
-      .sort({name: 1})
-      .cursor()
-      .on('data', function(service) {
-        var self = this;
-        self.pause();
-        RouteModel.io.find({
-          service: service._id,
-          public: true
-        }, function(err, routes) {
-          if (err)
-            self.destroy(err);
-          var serviceRoutes = [];
-          if (routes) {
-            routes.forEach(function(route) {
-              serviceRoutes.push({
-                id: route.routeId,
-                type: 'collections'
-              });
-              included.push({
-                id: route.routeId,
-                type: 'collections',
-                attributes: route.get('attributes'),
-                links: {
-                  self: res._apiuri + '/services/' + service.clientId + '/relationships/collections/' + route.routeId
-                },
-                meta: {
-                  creation: route.createdAt,
-                  modification: route.updatedAt
-                }
-              });
-            });
-          }
+      .paginate(query, options)
+      .then(function(result) {
+        result.docs.forEach(function(service) {
           var obj = service.getResourceObject(res._apiuri);
-          if (serviceRoutes.length > 0) {
-            obj.relationships = {
-              collections: {
-                links: {
-                  self: res._apiuri + '/services/' + service.clientId + '/relationships/collections',
-                  related: res._apiuri + '/services/' + service.clientId + '/collections'
-                },
-                data: serviceRoutes
-              }
-            };
-          }
+          var relationships = service.getRelationshipsObjects(res._apiuri);
+          included = _.union(included, relationships.included);
+          obj.relationships = relationships.relationships;
```

Comment:
TODO : move it into model 'getResourceObject' and create a new method 'getIncludedObjects'